backend/event-service/src/main/kotlin/com/ticketqueue/event/service/ScheduleService.kt

```
@@ -0,0 +1,174 @@
+
+    private fun initializeSeats(
+        schedule: EventSchedule,
+        seatTemplate: SeatTemplateDto,
+        priceByGrade: Map<SeatGrade, BigDecimal>
+    ): List<Seat> {
+        return seatTemplate.rows.flatMap { row ->
+            val gradeStr = seatTemplate.gradeMapping[row]
+                ?: throw EventException(ErrorCode.INVALID_SEAT_TEMPLATE_MAPPING)
+            val grade = try {
+                SeatGrade.valueOf(gradeStr)
+            } catch (e: IllegalArgumentException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)